packages/core/src/data-editor/use-autoscroll.ts

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import type { GridMouseCellEventArgs } from "../data-grid/data-grid-types";
+
+const maxPxPerMs = 2;
+const msToFullSpeed = 3000;
+
+export function useAutoscroll(
```

Comment:
Now, if you make the scroll direction a ref instead of state, you'll have to change how this hook works. Perhaps an approach is to only loop rAFs if you are dragging (or if the mouse is pressed?), and read the ref on every loop to figure out how much you need to move.